app/src/main/java/com/github/bgabriel998/softwaredevproject/points/UserPoint.java

```diff
@@ -29,24 +35,33 @@
     private boolean customLocation;
 
     /**
-     * Constructor for the UserPoint.
+     * Constructor for the UserPoint. Private because it is a singleton.
      *
      * @param mContext  the current context of the app
      */
-    public UserPoint(Context mContext) {
+    private UserPoint(Context mContext) {
         super(GPSTracker.DEFAULT_LAT, GPSTracker.DEFAULT_LON, GPSTracker.DEFAULT_ALT);
         gpsTracker = new GPSTracker(mContext, this);
         customLocation = false;
     }
 
-    public UserPoint(double lat, double lon, double alt){
-        super(lat,lon,alt);
+    /**
+     * Method to get the singleton instance of this class. If the class was already
+     * initialized the parameter will be ignored.
+     *
+     * @param mContext  context of the application.
+     * @return          single instance of the user point.
+     */
+    public static UserPoint getInstance(Context mContext) {
+        if (single_instance == null) {
+            return new UserPoint(mContext);
+        }
+        return single_instance;
```

Comment:
Can be changed for better readability:
```suggestion
        return single_instance == null ? new UserPoint(mContext) : single_instance;
```